.travis.yml

```diff
@@ -15,7 +15,22 @@ addons:
     chrome: stable
 install:
     - npm install
-# install synapse prerequisites for end to end tests
-    - sudo apt-get install build-essential python2.7-dev libffi-dev python-pip python-setuptools sqlite3 libssl-dev python-virtualenv libjpeg-dev libxslt1-dev
-script:
-    ./scripts/travis.sh
+    - ./scripts/travis/install-deps.sh
+
+matrix:
+    include:
+        - name: Linting Checks
```

Comment:
Are these steps run in parallel? On the same directory? Can they race?

Reply:
Is there a way to have an install step for all these sub-steps? That would be ideal ...

Reply:
Jobs are run in parallel, but in their own isolated containers so they don't interact with each other. You can tell travis to cache certain directories between runs, e.g. node-modules which is what `cache: npm` does, but that doesn't currently work for us due to the scripts editing node-modules in ways that are't portable.